src/bindings/go-elektra/CMakeLists.txt

```
@@ -0,0 +1,21 @@
+find_program (GO_EXECUTABLE go)
+
+if (GO_EXECUTABLE)
+	add_binding (go-elektra)
+
+	# build
+	add_custom_command (
+		OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/lib/libelektra.golib"
+		COMMAND ${CMAKE_COMMAND} ARGS -E env GO111MODULE=on PKG_CONFIG_PATH=${CMAKE_CURRENT_BINARY_DIR} ${GO_EXECUTABLE} build ./kdb
+		WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
```

```suggestion
		COMMAND ${CMAKE_COMMAND} ARGS -E env GO111MODULE=on PKG_CONFIG_PATH="${CMAKE_CURRENT_BINARY_DIR}" ${GO_EXECUTABLE} build ./kdb
		WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}")
```

Those are exactly the places where a space in the build directory path could cause a problem...

---

The error existed also before I added this…

---

Also `PKG_CONFIG_PATH=${CMAKE_CURRENT_BINARY_DIR}` won't help in any way AFAICT. There is no `elektra.pc` in this directory...

---

removed it!

---

> The error existed also before I added this…

It could also be that `elektra.pc` is simply generated incorrectly in `scripts/cmake/CMakeLists.txt`. Could you try building in a directory with a space and then check what the generated file `"build dir/scripts/cmake/elektra.pc"` looks like?

You probably also need to use something like `cmake -DCMAKE_INSTALL_PREFIX="$PWD/install" ..` to show the error.

---

Will try.

---

I have replaced my real `PWD` with `/my/path/to/libelektra/` here in this comment:

```text
prefix=/my/path/to/libelektra/build dir/install
exec_prefix=${prefix}/bin
libdir=${prefix}/lib
includedir=${prefix}/include/elektra
plugindir=${prefix}/lib/elektra
tool_execdir=${prefix}/lib/elektra/tool_exec
templatedir=${prefix}/share/elektra/templates
build_type=shared

Name: 
Description: 
Version: 0.9.14
Libs: -L${libdir} -lelektra-kdb -lelektra-core
Cflags: -I${includedir}
```

---

Yeah, that's what i suspected... Quotes won't work, but a `\` would:

```
prefix=/my/path/to/libelektra/build\ dir/install
```

So that means we need to fix the CMake script that generates this. Not sure how hard that is...

But I think this isn't actually what breaks in Jenkins. AFAICT Jenkins doesn't actually use a a subdirectory of the build dir as the `CMAKE_INSTALL_PREFIX`:

https://github.com/ElektraInitiative/libelektra/blob/d328680147620f5d78c00f547bf827287d2f6853/scripts/jenkins/Jenkinsfile#L60

So there is probably still some other place where the include dir is passed to the go stuff...

---

Just to note, it’s not just the include it also errors out for the linking library path.

i would suggest the following:
I will invest 1-2h more into finding a different solution, otherwise I would go with the solution I already committed yesterday (by not using a space in the dir name).

what do you say? Is this fine with you?

---

This is how the elektra.pc looks like which is used by elektrad which throws the error. Maybe fixing this `prefix=@CMAKE_BINARY_DIR@` helps? 

```
prefix=@CMAKE_BINARY_DIR@
libdir=${prefix}/lib
srcinclude=${prefix}/src/include
buildinclude=@CMAKE_SOURCE_DIR@/src/include

Name: @PACKAGE_NAME@
Description: @PACKAGE_DESCRIPTION@
Version: @KDB_VERSION@
Cflags: -I${srcinclude} -I${buildinclude}
Libs: -L${libdir} -lelektra-kdb -lelektr
```

---

I would leave it as it is now with the removed whitespace!